src/_artifacts.tf

```
@@ -0,0 +1,26 @@
+resource "massdriver_artifact" "api" {
+  field                = "api"
+  provider_resource_id = azurerm_communication_service.main.id
+  name                 = "Azure Communication Services ${azurerm_communication_service.main.name} (${azurerm_communication_service.main.id})"
```

Not just Azure! The `name`s everywhere are long and kinda gross. Users can change the name of an artifact _after_ it's been created, and further provisioning runs will _not_ update the name. So the first run is our best shot at something human-readable before they change it to "server number 5" or whatever they'd like.

TL;DR this is what this pattern looks like when rendered.

```
Virtual Network beebeaz-dev-central-b762 (/subscriptions/4718857a-5dbe-452c-a172-21a3ae74304f/resourceGroups/beebeaz-dev-central-b762/providers/Microsoft.Network/virtualNetworks/beebeaz-dev-central-b762)
```

vs

```
Virtual Network beebeaz-dev-central-b762 
```

The resource IDs in any cloud are v v long and they're passed as the `provider_resource_id` anyway. Personally, I think it looks better w/o it in the `name`.

```suggestion
  name                 = "Azure Communication Services ${azurerm_communication_service.main.name}"
```

---

Good point! I'm good with this change.